modin/experimental/backends/omnisci/query_compiler.py

```diff
@@ -619,17 +620,35 @@ def has_multiindex(self, axis=0):
         assert axis == 1
         return isinstance(self.columns, pandas.MultiIndex)
 
-    def get_index_name(self):
-        return self._modin_frame.get_index_name()
+    def get_index_name(self, axis=0):
+        return self.columns.name if axis else self._modin_frame.get_index_name()
```

Comment:
Any reason why we directly working with `columns`, but with `index` in another way?

Reply:
This is a trait of OmniSci backend, we can't access/modify indices without materialization of the whole frame (because index is a column inside a frame and to access its values we need to trigger execution), when columns are just labels, and OmniSci backend can work with them in a lazy mode